dali/pipeline/data/tensor_vector.cc

```diff
@@ -62,11 +63,112 @@ TensorVector<Backend>::TensorVector(TensorVector<Backend> &&other) noexcept {
   other.tensors_.clear();
 }
 
-
 template <typename Backend>
-size_t TensorVector<Backend>::nbytes() const noexcept {
+void TensorVector<Backend>::UnsafeSetSample(int dst, const TensorVector<Backend> &owner, int src) {
+  // TODO(klecki): more consistency checks, contiguous -> non-contiguous removes shares_data from
+  // samples
+  if (type() == DALI_NO_TYPE && owner.type() != DALI_NO_TYPE) {
+    set_type(owner.type());
+  }
+  if (!order()) {
+    set_order(owner.order());
+  }
```

Comment:
That's a footgun... The sequence `if (!order()) set_order(...)` is fine, but not in something that affects just one sample. I'm afraid it's another incarnation of a side-effect control flow, like allocation in `mutable_data`, which we got rid of.

Reply:
My intention was as following:
* we keep the source of through in batch object (for now type & order, next PR will move the shape, dimensionality etc)
* if the current order is empty (for the batch object), and someone sets a sample, we get the order from it and use it from now on.
* if we have some order already and we set a new sample, the order of new sample must match, so we don't have one order for batch and different one for samples.

So I can just make it hard error if the order that is currently set for batch doesn't match the one for the new sample?

Reply:
> we keep the source of *truth* (?) in batch object (for now type & order, next PR will move the shape, dimensionality etc)

What would it look like? We'd require the parent object to remain alive and unchanged? This is a big requrimeent that would invalidate any and all attempts at having an owning SampleView. Chasing pointers is not a good idea performancewise. Also, this would make the view object potentially inconsistent - we keep a _copy_ of a data pointer, so we should keep a _copy_ of the metadata - at least the most essential, which is shape and type.

Reply:
I moved away all of properties setup from the SetSample/CopySample